lib/StringToClassRule.php

```diff
@@ -37,7 +37,7 @@ public function processNode(Node $node, Scope $scope): array
             $className = \substr($className, 1);
         }
         $messages  = [];
-        if (! \preg_match('/^\\w.+\\w$/u', $className)) {
+        if (! \preg_match('/^\x5c\w.+\x5c\w$/u', $className)) {
```

Comment:
1. I'm sorry I don't get this suggestion: here we are searching for strings that start and end with a letter or digit as class names have this very constraint. The double backslash is an escape for the single-quoted string, not intended for the regex
2. The tests fail, and you hadn't add new tests: would exactly would you like to fix?

Reply:
@Slamdunk thank you for the reply. I misunderstood your indention, I thought you wanted to check only those strings, that have leading backslash and also at the end of strting - backslash and some word :) 

We have a problem because the rule checks for every string and for example `Error` string fails, although it's the internal class. I don't know why, but there is a check 
```php 
if ($classRef->isInternal() && $classRef->getName() !== $className) {
      return $messages;
}
```
So although `Error` is internal the second check `$classRef->getName() !== $className` is false.

Reply:
I guess it's a different issue and backslash has nothing to do with that :)